### ISSUE TITLE
Duty cycle command naming

### ISSUE THREAD

**Issue by ghost** [pret/pokered#113]
I am not sure if "duty" and "dutycycle" are the best names. They could be potentially confusing because they don't agree with the usual meaning of "duty cycle".

Normally, the term "duty cycle" means the percentage of time a signal is active over one cycle (on a Game Boy, this can be 12.5%, 25%, 50% or 75%).

We are calling a fixed duty cycle "duty", which is fine, but we have another command called "dutycycle" which seems like it should be a synonym but isn't.

The "dutycycle" command specifies 4 duty cycles. Every update, the duty cycles are rotated. This results in a sequence of duty cycles that repeats every 4 updates. It could be called a cycle of duty cycles.

I don't know what the best names would be. Maybe "dutycycle" and "dutycycles" would make it more obvious. I also think that the existing "dutycycle" command should take 4 parameters rather than one.

What do you think?

**Comment by Sanqui** [comment]
What you explain `dutycycle` as - a change of "duty cycle" in time - is commonly called _pulse width modulation_ in chiptune.

**Comment by dannye** [comment]
I've been thinking about this again lately and I want to rename `duty` to `dutycycle`.
Then the issue is what to rename the current `dutycycle`.
I think `dutycycles` or `dutycyclecycle` looks too confusing/similar, and I think `pwm` might disrupt the easy readability of the sound files.
I'm thinking about going with `dutycyclepattern` but then that's three words cramped together, so it would probably look better as `duty_cycle_pattern`.
But then we should probably change all two-word sound commands to use underscores, for consistency, like `stereopanning` -> `stereo_panning` etc.
And then we should probably update pokecrystal to match.

And while on the subject, I don't really like the names I originally chose for `callchannel`, `loopchannel`, and `endchannel`.
"channel" is not really appropriate in these contexts. But I didn't really want to just call them `call`, `loop`, and `end` because that's too vague. (can't use `call` anyway obviously)
I'm thinking about changing those to `call_branch`, `loop_sound`, and `end_branch` but not entirely satisfied with these either.
Also keep in mind that `endchannel` is used in multiple contexts. It's used to end a channel that doesn't loop, and it's also used to return from a branch that has been called with `callchannel`.
Also, the noise instrument names are pretty bad... not descriptive at all and some are just wrong.
I was very new when naming all this stuff :p

So basically what I'm saying is I think all audio macro names should be re-evaluated.
This is pretty much why I've been putting off this issue (for almost two years.. oh crap) cause I knew it would lead to this can of worms...

Anyone have any feedback?

**Comment by Rangi42** [comment]
"All audio macro names should be re-evaluated" is essentially [pokecrystal's issue #556](https://github.com/pret/pokecrystal/issues/556); hopefully the two repos can coordinate on better names.

**Comment by dannye** [comment]
Current suggestions:
- `pitchenvelope` -> `pitch_sweep`
- `squarenote` -> `square_note`
- `noisenote` -> `noise_note`
- `notetype` -> `note_type`
- `toggleperfectpitch` -> `toggle_perfect_pitch`
- `pitchbend` -> `pitch_bend`
- `duty` -> `duty_cycle`
- `stereopanning` -> `stereo_panning`
- `executemusic` -> `execute_music`
- `dutycycle` -> `duty_cycle_pattern`
- `callchannel` -> `call_sound`
- `loopchannel` -> `loop_sound`
- `endchannel` -> `ret_sound`

**Comment by dannye** [comment]
@Sanqui @Rangi42 Do you want to review #223 to see if I've missed anything?
I'll make similar updates to pokecrystal soon to make them as compatible as possible.
